parsl/executors/high_throughput/executor.py

```diff
@@ -524,6 +529,12 @@ def submit(self, func, *args, **kwargs):
         Returns:
               Future
         """
+        if len(resource_specification) > 0:
```

Comment:
Since resource spec is a dict, we can say `if resource_specification` since an empty dict evals to False.